src/pudl/analysis/record_linkage/model_helpers.py

```diff
@@ -1,18 +1,134 @@
+logger = pudl.logging_helpers.get_logger(__name__)
+
+
+def get_model_config(experiment_name: str) -> dict:
     """Load model configuration from yaml file."""
     config_file = (
         importlib.resources.files("pudl.package_data.settings")
         / "record_linkage_model_config.yml"
     )
     config = yaml.safe_load(config_file.open("r"))
 
-    if not (model_config := config.get(model_key)):
-        raise RuntimeError(f"No {model_key} entry in {config_file}")
+    if not (model_config := config.get(experiment_name)):
+        raise RuntimeError(f"No {experiment_name} entry in {config_file}")
+
+    return {experiment_name: model_config}
+
+
+def flatten_model_config(model_config: dict) -> dict:
+    """Take nested dictionary defining model config and flatten for logging purposes."""
+
+    def _flatten_level(config_level: dict, param_name: str):
+        flattened_dict = {}
+        for key, val in config_level.items():
+            flattened_param = f"{param_name}.{key}"
+            if isinstance(val, dict):
+                flattened_dict |= _flatten_level(val, param_name=flattened_param)
+            else:
+                flattened_dict[flattened_param[1:]] = val
+        return flattened_dict
+
+    return _flatten_level(model_config, "")
+
+
+class ExperimentTrackerConfig(Config):
+    """Dagster config to setup experiment tracking with mlflow."""
+
+    tracking_uri: str = f"sqlite:///{PudlPaths().output_dir}/experiments.sqlite"
+    tracking_enabled: bool = True
+    experiment_name: str
+    log_yaml: bool
+    run_context: str
+    #: Location to store artifacts. Artifact storage not currently used.
+    artifact_location: str = str(PudlPaths().output_dir)
+
+
+class ExperimentTracker(BaseModel):
+    """Class to manage tracking a single model."""
+
+    tracker_config: ExperimentTrackerConfig
+    run_id: str
+
+    @classmethod
+    def create_experiment_tracker(
+        cls, experiment_config: ExperimentTrackerConfig
+    ) -> "ExperimentTracker":
+        """Create experiment tracker for specified experiment."""
+        run_id = ""
+        if experiment_config.tracking_enabled:
+            mlflow.set_tracking_uri(experiment_config.tracking_uri)
+            with mlflow.start_run(
+                experiment_id=cls.get_or_create_experiment(
+                    experiment_name=experiment_config.experiment_name,
+                    artifact_location=experiment_config.artifact_location,
+                ),
+                tags={"run_context": experiment_config.run_context},
+            ) as run:
+                if experiment_config.log_yaml:
+                    config = flatten_model_config(
+                        get_model_config(experiment_config.experiment_name)
+                    )
+                    mlflow.log_params(config)
+
+                run_id = run.info.run_id
+
+        return cls(tracker_config=experiment_config, run_id=run_id)
+
+    def execute_logging(self, logging_func: Callable):
```

Comment:
Passing in a `Callable` here means that we can only execute it and do any logging when `tracking_enabled=True`, and the caller doesn't need to know whether tracking is enabled or not. This also means that tracking can be enabled/disabled from the dagster ui

Reply:
Oh interesting, so I could make a change, enable experiment tracking in the dagster UI, run that experiment, then make another change and disable experiment tracking all within the dagster UI?

Reply:
Yeah! We might end up just having it on all the time, but seemed like a nice to have